geode-core/src/test/java/org/apache/geode/internal/cache/ha/QueueRemovalMessageTest.java

```diff
@@ -0,0 +1,207 @@
+import org.apache.geode.cache.EntryNotFoundException;
+import org.apache.geode.cache.RegionDestroyedException;
+import org.apache.geode.distributed.internal.ClusterDistributionManager;
+import org.apache.geode.internal.cache.EventID;
+import org.apache.geode.internal.cache.HARegion;
+import org.apache.geode.internal.cache.InternalCache;
+
+public class QueueRemovalMessageTest {
+  private QueueRemovalMessage queueRemovalMessage;
+  private List<Object> messagesList;
+
+  private final ClusterDistributionManager dm = mock(ClusterDistributionManager.class);
+  private final InternalCache cache = mock(InternalCache.class);
+  private final String regionName1 = "region1";
+  private final String regionName2 = "region2";
+  private final HARegion region1 = mock(HARegion.class);
+  private final HARegion region2 = mock(HARegion.class);
+  private final HARegionQueue regionQueue1 = mock(HARegionQueue.class);
+  private final HARegionQueue regionQueue2 = mock(HARegionQueue.class);
+  private final EventID eventID1 = mock(EventID.class);
+  private final EventID eventID2 = mock(EventID.class);
+  private final EventID eventID3 = mock(EventID.class);
+
+
+  @Before
+  public void setup() {
+    queueRemovalMessage = spy(new QueueRemovalMessage());
+    messagesList = new LinkedList<>();
+    queueRemovalMessage.setMessagesList(messagesList);
+
+    long maxWaitTimeForInitialization = 30000;
+    when(cache.getRegion(regionName1)).thenReturn(uncheckedCast(region1));
+    when(cache.getRegion(regionName2)).thenReturn(uncheckedCast(region2));
+    when(region1.getOwnerWithWait(maxWaitTimeForInitialization)).thenReturn(regionQueue1);
+    when(region2.getOwnerWithWait(maxWaitTimeForInitialization)).thenReturn(regionQueue2);
+    when(regionQueue1.isQueueInitialized()).thenReturn(true);
+    when(regionQueue2.isQueueInitialized()).thenReturn(true);
+  }
+
+  @Test
+  public void messageProcessInvokesProcessRegionQueues() {
+    when(dm.getCache()).thenReturn(cache);
+
+    queueRemovalMessage.process(dm);
+
+    verify(queueRemovalMessage).processRegionQueues(eq(cache), any(Iterator.class));
+  }
+
+  @Test
+  public void processRegionQueuesCanProcessEachRegionQueue() {
+    addToMessagesList();
+    Iterator iterator = messagesList.iterator();
+
+    queueRemovalMessage.processRegionQueues(cache, iterator);
+
+    verify(queueRemovalMessage).processRegionQueue(iterator, regionName1, 1, regionQueue1);
+    verify(queueRemovalMessage).processRegionQueue(iterator, regionName2, 2, regionQueue2);
+    verify(queueRemovalMessage).removeQueueEvent(regionName1, regionQueue1, eventID1);
+    verify(queueRemovalMessage).removeQueueEvent(regionName2, regionQueue2, eventID2);
+    verify(queueRemovalMessage).removeQueueEvent(regionName2, regionQueue2, eventID3);
+  }
+
+  private void addToMessagesList() {
+    messagesList.add(regionName1);
+    messagesList.add(1);
+    messagesList.add(eventID1);
+    messagesList.add(regionName2);
+    messagesList.add(2);
```

Comment:
Can the two sizes here be extracted to constants, since they're referenced in multiple tests?